src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -174,11 +175,16 @@ class CoreWorkerDirectTaskSubmitter {
   absl::flat_hash_map<rpc::WorkerAddress, std::shared_ptr<rpc::CoreWorkerClientInterface>>
       client_cache_ GUARDED_BY(mu_);
 
-  /// Map from worker address to the lease client through which it should be
-  /// returned and its lease expiration time.
+  // max_tasks_in_flight_per_worker_ limits the number of tasks that can be pipelined to a worker using a single lease
+  const uint32_t max_tasks_in_flight_per_worker_;
+
+  /// Map from worker address to a tuple containing three elements:
+  /// - the lease client through which the worker should be returned
+  /// - the expiration time of the worker lease
+  /// - the number of tasks that are currently in flight to the worker
   absl::flat_hash_map<rpc::WorkerAddress,
-                      std::pair<std::shared_ptr<WorkerLeaseInterface>, int64_t>>
-      worker_to_lease_client_ GUARDED_BY(mu_);
+                      std::tuple<std::shared_ptr<WorkerLeaseInterface>, int64_t, uint32_t>>
```

Comment:
Can you make this a struct instead of a tuple? Sort of just a rule of thumb, but we generally make anything with more than 2 fields a struct for readability.